scip-java/src/main/scala/com/sourcegraph/scip_java/buildtools/ScipBuildTool.scala

```diff
@@ -154,22 +170,23 @@ class ScipBuildTool(index: IndexCommand) extends BuildTool("SCIP", index) {
     if (!Files.isDirectory(sourceroot)) {
       throw new NoSuchFileException(sourceroot.toString)
     }
-    val allSourceFiles = collectAllSourceFiles(sourceroot)
+    val allSourceFiles = collectAllSourceFiles(config, sourceroot)
     val javaFiles = allSourceFiles.filter(javaPattern.matches)
     val scalaFiles = allSourceFiles.filter(scalaPattern.matches)
     val kotlinFiles = allSourceFiles.filter(kotlinPattern.matches)
     if (javaFiles.isEmpty && scalaFiles.isEmpty && kotlinFiles.isEmpty) {
-      index
-        .app
-        .warning(
-          s"doing nothing, no files matching pattern '$sourceroot/**.{java,scala,kt}'"
-        )
+      if (config.sourceFiles.isEmpty)
```

Comment:
Hm, is it intentional that you are not using `allSourceFiles` here?

Reply:
Good catch, this is hacky. I changed the implementation to make the intention explicit: I don't want to report this message when indexing via Bazel because we generate a lot of scip-java.json files that may produce empty indexes, which is fine. I added a field to `Config.reportWarningOnEmptyIndex`, which is `true` by default but `false` when generated via the aspect.